src/Vizkit3dWorld.cpp

```diff
@@ -18,27 +18,37 @@
 
 namespace vizkit3d_world {
 
-Vizkit3dWorld::Vizkit3dWorld(std::string path, 
+Vizkit3dWorld::Vizkit3dWorld() {
+
+}
+
+Vizkit3dWorld::~Vizkit3dWorld()
+{
+}
+
+void Vizkit3dWorld::initialize(std::string path,
                             std::vector<std::string> modelPaths,
                             std::vector<std::string> ignoredModels,
-                            int cameraWidth, int cameraHeight, 
-                            double horizontalFov, 
+                            int cameraWidth, int cameraHeight,
+                            double horizontalFov,
                             double zNear, double zFar)
-    : worldPath(path)
-    , widget(NULL)
-    , modelPaths(modelPaths)
-    , app(NULL)
-    , cameraWidth((cameraWidth <= 0) ? 800 : cameraWidth)
-    , cameraHeight((cameraHeight <= 0) ? 600 : cameraHeight)
-    , zNear(zNear)
-    , zFar(zFar)
-    , horizontalFov(horizontalFov)
 {
+
+    worldPath = path;
+    widget = NULL;
+    modelPaths = modelPaths;
+    cameraWidth = (cameraWidth <= 0) ? 800 : cameraWidth;
+    cameraHeight = (cameraHeight <= 0) ? 600 : cameraHeight;
+    zNear = zNear;
+    zFar = zFar;
+    horizontalFov = horizontalFov;
+
     loadGazeboModelPaths(modelPaths);
 
     int argc = 1;
     char const*argv[] = { "vizkit3d_world" };
```

Comment:
QApplication requires that argc and argv remain valid for the entire lifetime of the QApplication instance, therefore you cannot declare them locally in this scope. 

Reference: http://doc.qt.io/qt-5/qcoreapplication.html#QCoreApplication

Reply:
The `QCoreApplication::instance()` is created in the main thread and I guess the orogen component is not running in the same thread. Therefore, it is necessary to instantiate QApplication for the same thread of orogen component. Also, I already made some tests using **qApp** macro and it failed.

Reply:
My point was that you are destroying the reference and the pointer you passed to QApplication because they are local in this scope, which is very likely what was causing the segmentation fault you were experiencing. About your comment, `QCoreApplication::instance()` does not create an instance, it returns the instance you created.

Reply:
> My point was that you are destroying the reference and the pointer you passed to QApplication because they are local in this scope, which is very likely what was causing the segmentation fault you were experiencing.
You could be right.

> QCoreApplication::instance() does not create an instance, it returns the instance you created.
I didn't know this. But we don't need access the reference, we just need to instantiate once. We are going to replace the static variable by `QCoreApplication::instance()`.

Reply:
👍